src/Formula/Printing.hs

```diff
@@ -142,14 +143,18 @@ instance Pretty PrologClause where
 instance Pretty PickInst where
   pretty  PickInst{..} =
       text "PickInst(" <> vcat
-                           [ nest 2 $ pretty cnfs
+                           [ nest 2 $ pretty formulas
                            , char ',' <+> pretty correct
                            , myText (", {" ++ show showSolution ++ "}")
                            , maybe empty (\s -> myText (", {" ++ show (toList s) ++ "}")) addText
                            , char ')'
                            ]
 
 
+instance Pretty FormulaInst where
+  pretty (InstCnf cnf) = text "CNF{" <> pretty cnf <> char '}'
+  pretty (InstDnf dnf) = text "DNF{" <> pretty dnf <> char '}'
+  pretty (InstArbitrary tree) = text "TREE{" <> pretty tree <> char '}'
```

Comment:
That seems in contradiction to less capitalized "Tree" in the Parser instance further above. So I would hope for a test failure.